packages/components/pie-modal/src/index.ts

```diff
@@ -33,15 +39,39 @@ export class PieModal extends RtlMixin(LitElement) {
         const headingTag = unsafeStatic(headingLevel);
 
         return html`
-            <dialog class="c-modal" ?open="${isOpen}">
-                <${headingTag} class="c-modal-heading">${heading}</${headingTag}>
-                <div class="c-modal-contentWrapper">
-                    <slot></slot>
-                </div>
+            <dialog id="dialog" class="c-modal" ?open="${isOpen}" @click="${this._handleCloseDialog}">
+                <header>
+                    <${headingTag} class="c-modal-heading">${heading}</${headingTag}>
+                    <pie-icon-button
+                        @click="${this._handleCloseDialog}"
+                        variant="ghost-tertiary"
+                        class="c-modal-closeBtn"></pie-icon-button>
+                </header>
+                <article>
+                    <div class="c-modal-contentWrapper">
+                        <slot></slot>
+                    </div>
+                </article>
             </dialog>
         `;
     }
 
+    _handleCloseDialog = () => {
+        this._dialog.close();
+        this._onDialogClose();
+    };
+
+    /**
+     * Dispatch `ON_MODAL_CLOSE_EVENT` event.
+     * To be used whenever we close the modal.
+     *
+     * @event
+     */
+    _onDialogClose = () => {
+        let event = new CustomEvent(ON_MODAL_CLOSE_EVENT);
```

Comment:
doesn't seem to be reassigned
```suggestion
        const event = new CustomEvent(ON_MODAL_CLOSE_EVENT);
```